pkg/cmd/pr/merge/merge.go

```diff
@@ -460,10 +460,22 @@ func (m *mergeContext) deleteRemoteBranch() error {
 	if !m.merged {
```

Comment:
I tried to improve the code around this `if`-statement, because it's confusing, but at the end gave up due to tests breaking. The coupling between the methods on `mergeContext`, and their behaviour against the state fields makes it very hard to understand/improve the code.